perfkitbenchmarker/benchmark_spec.py

```diff
@@ -229,7 +229,19 @@ def ConstructVirtualMachines(self):
 
         if DISK_SPEC in group_spec:
           disk_spec_class = disk.GetDiskSpecClass(cloud)
-          disk_spec = disk_spec_class(**group_spec[DISK_SPEC][cloud])
+          disk_spec_dict = group_spec[DISK_SPEC][cloud]
+          # We have to translate the data_disk_type flag here because
+          # this is where we know for sure what cloud we're using, and
+          # therefore we're able to pick the right translation table.
+          if FLAGS['data_disk_type'].present:
```

Comment:
@stfeng As far as I know, all you can do with FLAGS.data_disk_type is get its value. For anything else, you have to use FLAGS['data_disk_type'].

@ehankland I didn't do that because I didn't want to leave around an untranslated value in FLAGS.data_disk_type in case someone used it. Your comment made me realize that nobody uses it because it's brand new, and they shouldn't use it in any case. I will use your version.

This also resolves the issue of using `'data_disk_type'` instead of making a constant.